clients/src/main/java/org/apache/kafka/clients/producer/internals/RoundRobinPartitioner.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.producer.internals;
```

Comment:
Why do we add this to package `internals`?

Reply:
If having "Always RoundRobin" assignment isn't a desired internal option, would you say a new package "org.apache.kafka.clients.producer.external" would be appropriate?

Reply:
In general, the "internal" package is intended for things which are internal implementation details.  It's a bit odd to have the partitioner here since how it works will be part of a publicly described API, not an internal implementation detail.  Why don't we just put it in `clients/src/main/java/org/apache/kafka/clients/producer/` ?

Reply:
SGTM. Strictly speaking, it would require a KIP, as is adds new stuff to the public API. We might consider to move `DefaultPartitioner` out of `internals`, too.

Reply:
@cmccabe  and @mjsax - I am going to move this under producer as Colin suggested. I understand that it might require another KIP to move `DefaultPartitioner`. We are in need of this to go in the next release as our AWS deployment of Kafka requires this to be in place. We have been packaging this as a workaround within our application, and the PR will help us remove it out of application dependeny.

Thanks,

Reply:
> SGTM. Strictly speaking, it would require a KIP, as is adds new stuff to the public API

There is a KIP for this PR.  See [KIP-369: Alternative Partitioner to Support "Always Round-Robin" Selection.]*https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=89070828)

> We might consider to move DefaultPartitioner out of internals, too.

Maybe.  Users don't have to set any configuration to get DefaultPartitioner, so you could argue that the precise name and package of the default partitioner class is an implementation detail. :man_shrugging:

Reply:
Good that we have a KIP already. Loosing track... Too many KIPs at the same time.

Just looking into the KIP, it's not very explicit about the package name. The KIP should contain this information. I guess it's ok to just update the wiki page accordingly.

About `DefaultPartitioner`. Maybe. It's still a little weird that it is in package internal IMHO. But it's also not a big deal...